### ISSUE TITLE
Old status API fails for autosplitting task and breaks CRABMon

### ISSUE THREAD

**Issue by emaszs** [dmwm/CRABServer#5474]
This is an API on which CRABMon depends and apparently it doesn't work with autosplitting, which is to be expected. Might have to make a lightweight API for crabmon to get around this.

```
[27/Apr/2017:14:27:52]    Traceback (most recent call last):
[27/Apr/2017:14:27:52]      File "/data/srv/beHG1705d/sw/slc6_amd64_gcc493/cms/crabserver/3.3.1705.rc1-comp/lib/python2.7/site-packages/WMCore/REST/Server.py", line 1739, in dbapi_wrapper
[27/Apr/2017:14:27:52]        return handler(*xargs, **xkwargs)
[27/Apr/2017:14:27:52]      File "/data/srv/beHG1705d/sw/slc6_amd64_gcc493/cms/crabserver/3.3.1705.rc1-comp/lib/python2.7/site-packages/CRABInterface/RESTUserWorkflow.py", line 675, in get
[27/Apr/2017:14:27:52]        result = self.userworkflowmgr.status(workflow, verbose=verbose, userdn=userdn)
[27/Apr/2017:14:27:52]      File "/data/srv/beHG1705d/sw/slc6_amd64_gcc493/cms/crabserver/3.3.1705.rc1-comp/lib/python2.7/site-packages/CRABInterface/DataUserWorkflow.py", line 173, in status
[27/Apr/2017:14:27:52]        return self.workflow.status(workflow, userdn, userproxy, verbose=verbose)
[27/Apr/2017:14:27:52]      File "/data/srv/beHG1705d/sw/slc6_amd64_gcc493/cms/crabserver/3.3.1705.rc1-comp/lib/python2.7/site-packages/CRABInterface/Utils.py", line 186, in throttled_wrapped_function
[27/Apr/2017:14:27:52]        return fn(*args, **kw)
[27/Apr/2017:14:27:52]      File "/data/srv/beHG1705d/sw/slc6_amd64_gcc493/cms/crabserver/3.3.1705.rc1-comp/lib/python2.7/site-packages/CRABInterface/Utils.py", line 144, in wrapped_func
[27/Apr/2017:14:27:52]        return func(*args, **kwargs)
[27/Apr/2017:14:27:52]      File "/data/srv/beHG1705d/sw/slc6_amd64_gcc493/cms/crabserver/3.3.1705.rc1-comp/lib/python2.7/site-packages/CRABInterface/HTCondorDataWorkflow.py", line 683, in status
[27/Apr/2017:14:27:52]        job = int(job)
[27/Apr/2017:14:27:52]    ValueError: invalid literal for int() with base 10: '0SubJobs'
[27/Apr/2017:14:27:52] vocms0131.cern.ch 128.141.161.250 "GET /crabserver/preprod/workflow?workflow=170427_110314:oglez_crab_ogjob_2017_04_27_130244_electron_data2016g_3feb2017_v1_80VMINI HTTP/1.1" 500 Internal Server Error [data: 4994 in 
746 out 896313 us ] [auth: OK "/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=erupeika/CN=775659/CN=Emilis Antanas Rupeika" "" ] [ref: "" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:53.0) Gecko/20100101 Firefox/53.0" ]
```

**Comment by emaszs** [comment]
End result is that the Main tab is not displayed in the CRABMon page.

**Comment by emaszs** [comment]
Had not looked at this at all, still an issue.

**Comment by emaszs** [comment]
This is a bit annoying to fix since we've decided to get rid of the server-side status. I think we have to compromise somehow to keep the same functionality in CRABMon. In order to display the job list in the Main CRABMon page, I need to get it somehow, but this information is only available in the status_cache file. 

So far I see two options here - we either parse the status_cache file in the browser, similarly to how CRABClient does it except only to extract the jobList, or, we change the deprecated CRABServer status api and make it parse the status_cache file on the fly for each request to extract the jobList. OR I can drop the Main page in CRABMon entirely while only keeping the links, maybe the links are the only thing that's useful.

**Comment by belforte** [comment]
[Emilis, sorry for any spam, I sent by mail twice but those did not make it to github, so am pasting on the web UI]
I think the Main page is useful (at least to me) and if anything we can
review the DB fields shown there to avoid the need to "always" go
to the full DB view.
So restoring it even w/o the status, will be a good step forward.

About the condensed status summary, maybe we could create such a summary
along with the status_cache file and show that in the browser with minimal
formatting (maybe write an HTML to beign with ?). I'd rather not parse a
many-k item json file in the browser. 
In other words I find it very useful to see on this page the same summary a
crab status would print, w/o having to go through a crab remake step.

**Comment by belforte** [comment]
as indicated in https://github.com/dmwm/CRABServer/issues/5690#issuecomment-395378590
the problem is in `HTCondorDataWorkflow.status`

https://github.com/dmwm/CRABServer/blob/master/src/python/CRABInterface/HTCondorDataWorkflow.py#L452
and in particular at line
https://github.com/dmwm/CRABServer/blob/master/src/python/CRABInterface/HTCondorDataWorkflow.py#L683
since `job` is now a simply integer anymore in case of probe/tail jobs.

**Comment by belforte** [comment]
The way to reproduce is to call the REST API e.g. in the browser:

https://stefanovm.cern.ch/crabserver/dev/workflow?workflow=180829_183912:belforte_crab_20180829_203906

simply changing this line
https://github.com/dmwm/CRABServer/blob/master/src/python/CRABInterface/HTCondorDataWorkflow.py#L683
from
``` 
job = int(job)
```
to
```
 #job = int(job)
```
i.e. removig the conversion to integed makes the API not crash anymore [1] and restores display of main tab in UI

The information about running/pending/done jobs is now a bit misleading since it includes probe jobs ad does not differentialte processing, tails etc. pretty much like dashboard anyhow. So while not ideal, it is good enough for experts.

OTOH that only holds while probe jobs are running, once they are done code crash in
https://github.com/dmwm/CRABServer/blob/master/src/python/CRABInterface/HTCondorDataWorkflow.py#L677
with
`UnboundLocalError: local variable 'taskStatusCode' referenced before assignment`

[1] example from URL above:
```

{

    "desc": {
        "columns": [
            "tm_taskname",
            "panda_jobset_id",
            "tm_task_status",
            "tm_task_command",
            "tm_user_role",
            "tm_user_group",
            "tm_task_failure",
            "tm_split_algo",
            "tm_split_args",
            "panda_resubmitted_jobs",
            "tm_save_logs",
            "tm_username",
            "tm_user_dn",
            "tm_arguments",
            "tm_input_dataset",
            "tm_dbs_url",
            "tm_task_warnings",
            "tm_publication",
            "tm_user_webdir",
            "tm_asourl",
            "tm_asodb",
            "tm_output_dataset",
            "tm_collector",
            "tm_schedd",
            "tm_dry_run",
            "clusterid",
            "tm_start_time"
        ]
    },
    "result": [
        {
            "status": "SUBMITTED",
            "username": "belforte",
            "jobs": {
                "1": {
                    "State": "unsubmitted"
                },
                "2": {
                    "State": "unsubmitted"
                },
                "3": {
                    "State": "unsubmitted"
                },
                "4": {
                    "State": "unsubmitted"
                },
                "5": {
                    "State": "unsubmitted"
                },
                "0-4": {
                    "State": "failed",
                    "Error": [
                        8022,
                        "CmsRunFailure\nCMSSW error message follows.\nFatal Exception\nAn exception of category 'FatalRootError' occurred while\n   [0] Calling EventProcessor::runToCompletion (which does almost everything after beginJob and before endJob)\n   Additional Info:\n      [a] Fatal Root Error: @SUB=TTree::SetEntries\nTree branches have different numbers of entries, with 24000 maximum.\n",
                        {
                            "type": "Fatal Exception",
                            "details": "An exception of category 'FatalRootError' occurred while\n   [0] Calling EventProcessor::runToCompletion (which does almost everything after beginJob and before endJob)\n   Additional Info:\n      [a] Fatal Root Error: @SUB=TTree::SetEntries\nTree branches have different numbers of entries, with 24000 maximum.",
                            "exitCode": "8022"
                        }
                    ]
                },
                "0-5": {
                    "State": "failed",
                    "Error": [
                        8022,
                        "CmsRunFailure\nCMSSW error message follows.\nFatal Exception\nAn exception of category 'FatalRootError' occurred while\n   [0] Calling EventProcessor::runToCompletion (which does almost everything after beginJob and before endJob)\n   Additional Info:\n      [a] Fatal Root Error: @SUB=TTree::SetEntries\nTree branches have different numbers of entries, with 43200 maximum.\n",
                        {
                            "type": "Fatal Exception",
                            "details": "An exception of category 'FatalRootError' occurred while\n   [0] Calling EventProcessor::runToCompletion (which does almost everything after beginJob and before endJob)\n   Additional Info:\n      [a] Fatal Root Error: @SUB=TTree::SetEntries\nTree branches have different numbers of entries, with 43200 maximum.",
                            "exitCode": "8022"
                        }
                    ]
                },
                "0-1": {
                    "State": "failed",
                    "Error": [
                        8022,
                        "CmsRunFailure\nCMSSW error message follows.\nFatal Exception\nAn exception of category 'FatalRootError' occurred while\n   [0] Calling EventProcessor::runToCompletion (which does almost everything after beginJob and before endJob)\n   Additional Info:\n      [a] Fatal Root Error: @SUB=TTree::SetEntries\nTree branches have different numbers of entries, with 46800 maximum.\n",
                        {
                            "type": "Fatal Exception",
                            "details": "An exception of category 'FatalRootError' occurred while\n   [0] Calling EventProcessor::runToCompletion (which does almost everything after beginJob and before endJob)\n   Additional Info:\n      [a] Fatal Root Error: @SUB=TTree::SetEntries\nTree branches have different numbers of entries, with 46800 maximum.",
                            "exitCode": "8022"
                        }
                    ]
                },
                "0-2": {
                    "State": "running"
                },
                "0-3": {
                    "State": "failed",
                    "Error": [
                        8022,
                        "CmsRunFailure\nCMSSW error message follows.\nFatal Exception\nAn exception of category 'FatalRootError' occurred while\n   [0] Calling EventProcessor::runToCompletion (which does almost everything after beginJob and before endJob)\n   Additional Info:\n      [a] Fatal Root Error: @SUB=TTree::SetEntries\nTree branches have different numbers of entries, with 21000 maximum.\n",
                        {
                            "type": "Fatal Exception",
                            "details": "An exception of category 'FatalRootError' occurred while\n   [0] Calling EventProcessor::runToCompletion (which does almost everything after beginJob and before endJob)\n   Additional Info:\n      [a] Fatal Root Error: @SUB=TTree::SetEntries\nTree branches have different numbers of entries, with 21000 maximum.",
                            "exitCode": "8022"
                        }
                    ]
                },
                "0SubJobs": {
                    "State": "unsubmitted"
                }
            },
            "submissionTime": 1535567952,
            "jobdefErrors": [ ],
            "collector": "cmsgwms-collector-global.cern.ch,cmssrv221.fnal.gov",
            "jobList": [
                [
                    "failed",
                    "0-4"
                ],
                [
                    "failed",
                    "0-5"
                ],
                [
                    "failed",
                    "0-1"
                ],
                [
                    "running",
                    "0-2"
                ],
                [
                    "failed",
                    "0-3"
                ],
                [
                    "unsubmitted",
                    "0SubJobs"
                ],
                [
                    "unsubmitted",
                    "1"
                ],
                [
                    "unsubmitted",
                    "3"
                ],
                [
                    "unsubmitted",
                    "2"
                ],
                [
                    "unsubmitted",
                    "5"
                ],
                [
                    "unsubmitted",
                    "4"
                ]
            ],
            "publication": { },
            "schedd": "crab3@vocms0198.cern.ch",
            "taskWarningMsg": [ ],
            "jobsPerStatus": {
                "failed": 4,
                "running": 1,
                "unsubmitted": 6
            },
            "publicationFailures": { },
            "statusFailureMsg": "",
            "taskFailureMsg": "",
            "failedJobdefs": 0,
            "command": "SUBMIT",
            "ASOURL": "https://cmsweb.cern.ch/crabserver/prod",
            "totalJobdefs": 0,
            "outdatasets": null
        }
    ]

}

```

**Comment by belforte** [comment]
the latter problem is due to `taskStatusCode` not being defined unless one uses old logic, an error already spotted by e.g. PyCharm.
So I am making sure that it is defined.
All in all this API does not need to be perfect, the only aim here is that the UI still shows the main tab where useful information is printed for the operator as soon as UI is entered.
